pkg/docker/containers-view.jsx

```diff
@@ -185,14 +209,59 @@ var ContainerList = React.createClass({
         this.setState({ containers: containers });
     },
 
+    setNewProblem: function (c_id, url, message) {
+        /* New problem is always displayed, no matter if the same problem is
+         * already shown. It is because user may be interested into dynamic
+         * watching of the problems occurring. After refreshing the site, only
+         * the latest occurrence is displayed.
+         */
+        var known_problems = this.state.problems;
+        if ( c_id in known_problems )
+            known_problems[c_id].push([url, message]);
+        else
+            known_problems[c_id] = [[url, message]];
+        this.setState({ problems: known_problems });
+    },
+
+    newProblemOccurred: function (event, problem_path) {
+        var entry = this.problems_client.proxy('org.freedesktop.Problems2.Entry', problem_path);
```

Comment:
This proxy never gets removed, or does it? The [API docs](http://cockpit-project.org/guide/latest/cockpit-dbus.html) don't have a proxy close method similar to `client.close()`; @stefwalter, @petervo, am I missing anything?

I'm a bit cautious of creating a potentially large number of D-Bus proxies as they already keep hitting us on the /system/logs page. And here it's only being used for a single function call, which doesn't need all the overhead of proxies (like property watching). Perhaps you can just use a normal `cockpit.dbus.call()` here?

Reply:
I've been told by @petervo that the proxy closing is handled automatically.

Reply:
The proxy closing is handled automatically when the page is unloaded. But that still means we can have vast amounts of open proxies while the page is open. So it's worth making sure to close them when not in use, or not to use them at scale at all.

Reply:
So should I use `cockpit.dbus.call()`? If so, can you give me hint, how the current line can be changed?

Reply:
you should use client.call it's documented here. http://cockpit-project.org/guide/latest/cockpit-dbus.html

To get an object without getting a proxy you need to call "Get" or "GetAll" on the "org.freedesktop.DBus.Properties" for your path to populate the object.

You can see an example of code doing something similar here: https://github.com/cockpit-project/cockpit/blob/master/pkg/lib/service.js#L192

Then your method calls would also need to use client.call in this case you can just call the method name of your interface directly.